### PR TITLE
Added condition attributes to schema and invoke-plaster for parameters.

### PR DESCRIPTION
**Pull request by zloeber**
Added schema changes to allow for the condition attribute with parameter definitions. Included condition processing in Invoke-Plaster for parameters. I didn't update the schema version as I've been told that the schema version is already bumped up for some other changes that are underway.

### CODE REVIEW

**Comment by rkeithhill** [comment]
I've already been down this road and it didn't work out.  Conditional parameters do not work with dynamic parameters since you can specify a parameter that a later parameter's value would not allow via a condition on the first parameter that tests the value of the later parameter.

**Comment by zloeber** [comment]
Would it make sense to add an additional attribute like 'interactiveonly' which will cause the parameter to be skipped when generating the dynamic parameters?

**Comment by rkeithhill** [comment]
@zloeber I made a run at making the dynamic parameters work but ran into several problems.  One was that we don't create the constrained runspace when dynamic parameters are being processed.  I was able to solve that.  The show stopper for me was that I couldn't retrieve the argument value for already specified parameters.  If I could have gotten that then the dynamic parameters completion could have have been "really" dynamic.  :-)  Oh well.  

At this point, I still want this capability and I'm happy with being to able to "explore" further the issues with this feature and dynamic parameters.  Could you add notes to the two `condition` annotation elements along the lines of `This condition will not be evaluated during dynamicparam processing`?  Then I'll merge the PR.  

Sorry for the delay on this.  Between my experimentation with dynamic parameters, two birthdays, Mother's day, one HS commencement, several graduation parties, end of school year performances and a kiddo in the E.R. - May has been a **crazy** month!

FWIW the conclusion I've come to is that while someone can specify an invalid combination of parameters & arg values, they will be able to specify a `valid` combination.  To help, I think I would like to add a feature (switch param) to Invoke-Plaster (or maybe a new command) where you can answer all the template questions and get a hashtable that can be used for splatting.

**Comment by zloeber** [comment]
I too have been trying to get multilevel dynamic parameters to work and was only ever able to get it to work using C# (and even then, only with reflection hacks). I have a whole get-unboundparameter powershell translated function and everything that I simply couldn't get to return the correct data (via $PSCmdlet and the aforementioned .Net reflection). I've finally thrown the towel in on that one as it was such a crazy work around that, even if it functioned, it would have felt wrong to use :) 

As for your last statement, would it be wrong to include that data with the -Passthrough switch?

File: src/Schema/PlasterManifest-v1.xsd [hunk 1]
```diff
@@ -175,6 +175,11 @@
           <xs:documentation>The label for the choice that is presented in the user interface. The &amp; character must be respresented with the entity ref &amp;amp;</xs:documentation>
         </xs:annotation>
       </xs:attribute>
+      <xs:attribute name="condition" type="ptd:condition">
```
Comment (rkeithhill) [inline src/Schema/PlasterManifest-v1.xsd]: A conditional choice is cool (not something I had thought of before) but I don't see that you are checking this condition where `choice|multichoice` is being processed.  BTW sorry about that other PR mixup with @bgelens, I should have had that accepted a while back.  Thanks for your work on it!   If you're running out of gas on this one, I can accept it and add the `choice` check but I wanted to give you a chance to wrap this one up.  :-)

Comment (zloeber) [inline src/Schema/PlasterManifest-v1.xsd]: Sorry, I didn't mean to drop off on this one. Thanks for tying this one off (and giving opportunities to allow me to complete it as well).

Comment (rkeithhill) [inline src/Schema/PlasterManifest-v1.xsd]: No problem.  Thanks for the contribution!

**Pull request merged**